### ISSUE TITLE
Build failure: devcontainer

### ISSUE THREAD

**Issue by seanrmurphy** [NixOS/nixpkgs#314644]
### Steps To Reproduce

```
$ uname -a
Linux nixos 6.9.1 #1-NixOS SMP PREEMPT_DYNAMIC Fri May 17 10:18:09 UTC 2024 x86_64 GNU/Linux
$ nix-build --check -A devcontainer
```

### Build log

```
checking outputs of '/nix/store/06kqng7kvgd9vlyz229kww4jlqfpr4ii-devcontainer-0.60.0.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/axrcfrdihcc90v1sfkfv3s0f6sbkk5p1-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
yarn config v1.22.22
success Set "yarn-offline-mirror" to "/nix/store/7q311bg1n8v5zgbrkjxx336zycm38ff9-offline".
Done in 0.03s.
yarn install v1.22.22
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
[] 0/554[] 16/554[] 30/554[] 51/554[] 76/554[] 100/554[] 112/554[] 141/554[] 157/554[] 185/554[] 190/554[] 195/554[] 197/554[] 202/554[] 205/554[] 207/554[] 223/554[] 251/554[] 291/554[] 324/554[] 351/554[] 382/554[] 415/554[] 442/554[] 475/554[] 505/554[] 530/554[] 550/554[] 551/554[] 553/554[4/5] Linking dependencies...
[] 554/554[] 0/589[] 0/16489[] 1059/16489[] 2380/16489[] 3929/16489[] 5653/16489[] 7856/16489[] 9257/16489[] 10900/16489[] 13252/16489[] 15246/16489[] 16467/16489[] 0/1017[5/5] Building fresh packages...

error /build/source/node_modules/esbuild: Command failed.
Exit code: 126
Command: node install.js
Arguments: 
Directory: /build/source/node_modules/esbuild
Output:
/bin/sh: node: Text file busy
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

error: builder for '/nix/store/06kqng7kvgd9vlyz229kww4jlqfpr4ii-devcontainer-0.60.0.drv' failed with exit code 126;
       last 10 log lines:
       > 
       > error /build/source/node_modules/esbuild: Command failed.
       > Exit code: 126
       > Command: node install.js
       > Arguments:
       > Directory: /build/source/node_modules/esbuild
       > Output:
       > /bin/sh: node: Text file busy
       > info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
       > 
       For full logs, run 'nix log /nix/store/06kqng7kvgd9vlyz229kww4jlqfpr4ii-devcontainer-0.60.0.drv'.
```

### Additional context

- Kernel version: 6.9.1
- Was looking into this as part of the ZHF - just took this package to look into it
- https://hydra.nixos.org/build/260969645 

Seems like this is a very specific issue and it's related to this:
https://github.com/nodejs/docker-node/issues/1912

I found that when I dropped down to a lower kernel version (6.1.91), the problem disappeared. I guess that when the kernel version changes, this problem will just go away.

I spent a little time looking into using different node versions, but this did not solve the problem easily; it seems that `yarn` prob ensures a particular `node` version for at least some dependencies.

**Comment by seanrmurphy** [comment]
Pretty sure this is the same issue as reported in #314713 which seems to be fixed with 6.9.3 kernel.

**Comment by seanrmurphy** [comment]
`devcontainer` is now building properly on hydra so it appears that this issue has been resolved:

https://hydra.nixos.org/job/nixpkgs/trunk/devcontainer.x86_64-linux
